code/modules/mob/living/simple_animal/bot_swarm/swarmer.dm

```diff
@@ -0,0 +1,598 @@
+////Deactivated swarmer shell////
+/obj/item/unactivated_swarmer
+	name = "unactivated swarmer"
+	desc = "A currently unactivated swarmer. Swarmers can self activate at any time, it would be wise to immediately dispose of this."
+	icon = 'icons/mob/swarmer.dmi'
+	icon_state = "swarmer_unactivated"
+
+/obj/item/unactivated_swarmer/New()
+	notify_ghosts("An unactivated swarmer has been created in [get_area(src)]! <a href=?src=\ref[src];ghostjoin=1>(Click to enter)</a>")
+	..()
+
+/obj/item/unactivated_swarmer/Topic(href, href_list)
+	if(href_list["ghostjoin"])
+		var/mob/dead/observer/ghost = usr
+		if(istype(ghost))
+			attack_ghost(ghost)
+
+/obj/item/unactivated_swarmer/attack_ghost(mob/user)
+	var/be_swarmer = alert("Become a swarmer? (Warning, You can no longer be cloned!)",,"Yes","No")
+	if(be_swarmer == "No")
+		return
+	if(qdeleted(src))
+		user << "Swarmer has been occupied by someone else."
+		return
+	var/mob/living/simple_animal/hostile/swarmer/S = new /mob/living/simple_animal/hostile/swarmer(get_turf(loc))
+	S.key = user.key
+	qdel(src)
+
+////The Mob itself////
+
+/mob/living/simple_animal/hostile/swarmer
+	name = "Swarmer"
+	real_name = "Swarmer"
+	icon = 'icons/mob/swarmer.dmi'
+	desc = "A robot of unknown design, they seek only to consume materials and replicate themselves indefinitely."
+	speak_emote = list("tones")
```

Comment:
Considering these guys are definitely not your standard silicons, it would be best if it wasn't 'states', especially since machines use that as well; again, I don't think "tones" sounds that bad; it implies beeping and booping and telephony based sounds.